### ISSUE TITLE
callback after permission granted (android 6)

### ISSUE THREAD

**Issue by meceo** [EddyVerbruggen/nativescript-barcodescanner#26]
Currently in order to activate a scanner I have following code:

``` javascript
barcodescanner.hasCameraPermission().then(function(granted) {
  if (granted) {
    activateScanner();
  } else {
    barcodescanner.requestCameraPermission().then(function() {
      console.log("called before permission granted!");
      return activateScanner();
    });
  };
});
```

Please notice callback for requestCameraPermission fires before permission is granted or denied. Is that in purpose? If so, is there any other callback I can use to activate scanner after user grands permission?

**Comment by meceo** [comment]
sorry for my laziness, now when I read your code I see the comment: https://github.com/EddyVerbruggen/nativescript-barcodescanner/blob/master/barcodescanner.android.js#L32

do you plan to change behaviour anytime soon?

**Comment by EddyVerbruggen** [comment]
Hey @meceo yes I do but I need to figure out if it's possible first. It wasn't at the time but it might well be now.

**Comment by EddyVerbruggen** [comment]
I don't like the dependency on support-v4, but this seems like the best solution for me for now: https://github.com/NathanaelA/nativescript-permissions.. I'm probably going to add it as a dependency rather than copy-pasting the core code (which is similar to [this](https://github.com/NativeScript/nativescript-geolocation/blob/master/source/nativescript-geolocation.android.ts) as well).

**Comment by creambyemute** [comment]
Would like to have this as well :)

**Comment by EddyVerbruggen** [comment]
Available in 1.5.0. It's not on npm yet as I'd like you to test it first by installing the plugin from github. That will also give you the ability to toggle the torch through the hardware volume buttons.
